djangocms_navigation/admin.py

```diff
@@ -59,10 +71,191 @@ class MenuContentAdmin(admin.ModelAdmin):
+
+    def get_version(self, obj):
+        """
+        Return the latest version of a given object
+        :param obj: MenuContent instance
+        :return: Latest Version linked with MenuContent instance
+        """
+        return obj.versions.all()[0]
+
+    def get_versioning_state(self, obj):
+        """
+        Return the current state of a version (DRAFT, PUBLISHED or ARCHIVED)
+        :param obj: Version instance
+        :return: Version status
+        """
+        return self.get_version(obj).get_state_display()
+
+    get_versioning_state.short_description = _("Lock Status")
+
+    def is_locked(self, obj):
+        """
+        Render the lock item if the item is in a draft state and version locked
+        :param obj: MenuContent instance
+        :return: string to render
+        """
+        version = self.get_version(obj)
+        if not using_version_lock:
+            return ""
+        elif version.state == DRAFT and version_is_locked(version):
+            return render_to_string("djangocms_version_locking/admin/locked_icon.html")
+        return ""
+
+    is_locked.short_description = _("Lock Status")
+
+    def get_state_display(self, obj):
+        """
+        Return the current state of the version
+        :param obj: MenuContent Instance
+        :return: State
+        """
+        return self.get_version(obj).get_state_display()
+
+    get_state_display.short_description = _("State")
+
+    def get_author(self, obj):
+        """
+        Return the author who created a version
+        :param obj: MenuContent Instance
+        :return: Author
+        """
+        return self.get_version(obj).created_by
+
+    get_author.short_description = _("Author")
+
+    def get_modified_date(self, obj):
+        """
+        Get the last modified date of a version
+        :param obj: MenuContent Instance
+        :return: Modified Date
+        """
+        return self.get_version(obj).modified
+
+    get_modified_date.short_description = _("Modified")
+
+    def _list_actions(self, request):
+        """
+        A closure that makes it possible to pass request object to
+        list action button functions.
+        """
+
+        def list_actions(obj):
+            """Display links to state change endpoints
+            """
+            return format_html_join(
+                "",
+                "{}",
+                ((action(obj, request),) for action in self.get_list_actions()),
+            )
+
+        list_actions.short_description = _("actions")
+        return list_actions
+
+    def get_list_actions(self):
+        return [
+            self._get_preview_link,
+            self._get_edit_link,
+        ]
+
+    def get_list_display(self, request):
+        menu_content_list_display = ["title"]
+        versioning_enabled = is_versioning_enabled(self.model)
+
+        if versioning_enabled:
```

Comment:
I think we need to revise this so that the actions are listed at the end the same as the page admin.

Also the rules for version lock appear to control all of the versioning items too. I think this should probably be:

```
if versioning_enabled:
    menu_content_list_display.extend(["get_versioning_state", "get_author", "get_modified_date", "get_state_display"]
    # Add version locking specific items
    if using_version_lock:
        menu_content_list_display.extend(["is_locked"])
    # Ensure actions are the last items
    menu_content_list_display.extend([self._list_actions(request)])
else:  
    menu_content_list_display.extend(["get_menuitem_link", "get_preview_link"])
     


```